### ISSUE TITLE
AddHoliday does not support year

### ISSUE THREAD

**Issue by cbelsole** [rickar/cal#47]
We are currently removing and adding holidays to our calendar which supports the last ~4 years. We don't want to instantiate a new calendar for every year so our options look like:

```diff
	calendar.AddHoliday(
                cal.USIndependence,
                cal.USLabor,
                cal.USThanksgiving,
-               cal.NewHoliday(time.December, 24),
+               cal.USColumbus,
                cal.USChristmas,
        )
```

So our options now look like:

```go
	calendar.AddHoliday(
                cal.USIndependence,
                cal.USLabor,
                cal.USThanksgiving,
                cal.USChristmas,
        )

	cal.NewHolidayExact(time.December, 24, 2017)
	cal.NewHolidayExact(time.December, 24, 2018)
	cal.NewHolidayExact(time.December, 24, 2019)
        // Columbus Day
	cal.NewHolidayExact(time.October, 12, 2020)
```

This would keep the calendar passive for the last few years for Christmas Eve and update the calendar for 2020 for Columbus day. This is not elegant as it forces us to update the code every year with the new holidays for that year. Something like this instead would be more elegant:

```go
	calendar.AddHoliday(
                cal.USIndependence,
                cal.USLabor,
                cal.USThanksgiving,
                cal.NewHoliday(time.December, 24).UpTo(2019),
                cal.USColumbus.StartingAt(2020),
                cal.USChristmas,
        )
```

This should be a passive change to the library. So it could work for v1 and v2.

**Comment by rickar** [comment]
If I understand what you're trying to do, v2 already supports this with Holiday.StartYear, Holiday.EndYear, and Holiday.Except.

**Comment by cbelsole** [comment]
It looks like `StartYear` and `EndYear` fulfill our requirements. I have a piece of feedback after implementing it. I had to write code like this:

```go
columbusDay := us.ColumbusDay.Clone(us.ColumbusDay.Name, us.ColumbusDay.Type, us.ColumbusDay.Observed)
columbusDay.StartYear = 2020

veteransDay := us.VeteransDay.Clone(us.VeteransDay.Name, us.VeteransDay.Type, us.VeteransDay.Observed)
veteransDay.StartYear = 2020

calendar.AddHoliday(
	columbusDay,
	veteransDay,
)
```

It would be nice if `Clone()` took something like:
```go
CloneOptions{
	Name *string
	// ... and the rest of the fields as nullable fields
}
```

Then inside clone it would resolve the clone options with the original holiday. That way I could write code like:
```
calendar.AddHoliday(
	us.ColumbusDay.Clone(cal.CloneOptions{StartYear: 2020}),
	us.VeteransDay.Clone(cal.CloneOptions{StartYear: 2020}),
)

**Comment by rickar** [comment]
Great idea, thanks! I implemented that in the latest release (v2.0.0-beta.2).
